Oracle.md

```diff
@@ -0,0 +1,165 @@
+# Oracle specifications
+
+## Introduction
+
+For the purpose of these specifications, an event is a digital representation of a real world fact.
+An [oracle](./Introduction.md#Oracle) is an entity that commits to publishing a signature over a (number of) event outcome(s) ahead of time by releasing an [R-value](./Introduction.md#R-value) as well as necessary information for two parties to build a set of [CETs](./Introduction.md#Contract-Execution-Transaction-(CET)).
+This necessary information is committed in a so-called _event descriptor_ that will be further detailed in this document.
+Descriptors are classified and discovered through a namespace presently represented as Unique Resource Identifier (URI) paths.
+
+## Event descriptor
+
+An event descriptor provides information to clients about an event for which an oracle plans on releasing a signature over its outcome.
+The provided information should be sufficient for a client to create a set of CETs that will cover all possible outcomes of the event.
+
+Here we assume that an event outcome can be represented either as a set of strings or numbers.
+Four kinds of event outcomes are defined (note that a single even can be composed of several types).
+
+### External event descriptors
+
+External event descriptor enables representation of oracles in a format agreed upon between an oracle and its clients, for the cases where the following descriptor types are inadequate.
```

Comment:
May add a suggestion that this is useful for testing but that oracle developers should prefer to PR new types into the spec rather than using this when possible

Reply:
added